lib/puppet/application/agent.rb

```diff
@@ -387,9 +388,19 @@ def onetime(daemon)
 
   def main(daemon)
     Puppet.notice _("Starting Puppet client version %{version}") % { version: Puppet.version }
+    Puppet.debug {"Current config:\n#{get_config}"}
     daemon.start
```

Comment:
I'm not seeing `Starting...` nor `Current config` in the output when running with `bundle exec puppet agent -d --logdest /tmp/puppet.log`. I'm assuming this is a problem with daemonizing and log destinations. The agent is supposed to fork and the parent exits while the child process calls `Puppet::Daemon.close_streams` to reopen its stdout, etc file descriptors and log destinations.

Reply:
I don't remember how I tested this, I assume it was in a fully working environment, as now looks to exit earlier as no server is configured.
I will move this earlier in startup, as earlier should be more useful.